src/scry/completion/method_db.cr

```diff
@@ -53,39 +57,54 @@ module Scry::Completion
 
   class Visitor < Crystal::Visitor
     property classes
+    property class_name = ""
 
     def initialize(@file : String)
       @classes = {} of String => Array(MethodDBEntry)
-      @class_queue = [] of String
+      @class_queue = [] of Crystal::Path
+      @module_queue = [] of Crystal::Path
+    end
+
+    def generate_name(paths : Array(Crystal::Path))
+      paths.flat_map(&.names).compact.join("::")
     end
 
     def visit(node : Crystal::Def)
       return false if @class_queue.empty? || node.visibility != Crystal::Visibility::Public
       if node.name == "initialize"
         name = "new"
-        method_receiver = "#{@class_queue.last}.class"
-        return_type = @class_queue.last
+        method_receiver = "#{@class_name}.class"
+        return_type = @class_name
       else
         name = node.name
-        method_receiver = node.receiver ? "#{@class_queue.last}.class" : @class_queue.last
+        method_receiver = node.receiver ? "#{@class_name}.class" : @class_name
       end
       signature = "(#{node.args.map(&.to_s).join(", ")}) : #{return_type || node.return_type.to_s}"
       @classes[method_receiver] << MethodDBEntry.new(name, signature, @file, node.location.to_s)
       false
     end
 
     def visit(node : Crystal::ClassDef)
-      @classes["#{node.name.to_s}.class"] = [] of MethodDBEntry
-      @classes[node.name.to_s] = [] of MethodDBEntry
-      @class_queue << node.name.to_s
+      @class_queue << node.name
+      @class_name = generate_name(@module_queue + @class_queue)
```

Comment:
Does this still work if you have a module inside a class inside a module?

Why are you separating module & classes queues?

Reply:
oh, I didn't even know that could happen

Reply:
It's seperate cause we only store methods for class, for now.

Reply:
@bew I tried to change this to use classes and modules in the same queue but started getting some tests failing not really sure why. I cut a new branch to add this and module functions as well.

Reply:
can you create a new issue to track and not forget this? It's quite important
thanks

Reply:
Also, include what you tried and your results, it might help find the actual problem ;)